WordPress/Classes/Services/ThemeService.m

```diff
@@ -0,0 +1,270 @@
+#import "ThemeService.h"
+
+#import "Blog.h"
+#import "RemoteTheme.h"
+#import "Theme.h"
+#import "ThemeServiceRemote.h"
+#import "WPAccount.h"
+
+@implementation ThemeService
+
+#pragma mark - Themes availability
+
+- (BOOL)accountSupportsThemeServices:(WPAccount *)account
+{
+    NSParameterAssert([account isKindOfClass:[WPAccount class]]);
+    
+    return [account isWPComAccount];
+}
+
+- (BOOL)blogSupportsThemeServices:(Blog *)blog
+{
+    NSParameterAssert([blog isKindOfClass:[Blog class]]);
+    
+    return blog.restApi && [blog dotComID];
+}
+
+#pragma mark - Local queries: Creating themes
+
+/**
+ *  @brief      Creates and initializes a new theme with the specified theme Id in the specified
+ *              context.
+ *  @details    You should probably not call this method directly.  Please read the documentation
+ *              for findOrCreateThemeWithId: first.
+ *
+ *  @param      themeId     The ID of the new theme.  Cannot be nil.
+ *
+ *  @returns    The newly created and initialized object.
+ */
+- (Theme *)newThemeWithId:(NSString *)themeId
+{
+    NSParameterAssert([themeId isKindOfClass:[NSString class]]);
+    
+    NSEntityDescription *entityDescription = [NSEntityDescription entityForName:[Theme entityName]
+                                                         inManagedObjectContext:self.managedObjectContext];
+    
+    Theme *theme = [[Theme alloc] initWithEntity:entityDescription
+                  insertIntoManagedObjectContext:self.managedObjectContext];
```

Comment:
I'm fixing this with `performBlockAndWait:` for the time being, so I don't have to start adding success blocks everywhere in this PR.  Not the ideal but it can be improved separately if it proves to cause trouble and it's a way to avoid large modifications to this already-large PR.